test/utils/Deployers.sol

```diff
@@ -159,6 +165,23 @@ contract Deployers {
         uninitializedNativeKey.fee = 100;
     }
 
+    /// @notice Helper function for a simple swap that allows for unlimited price impact
+    function swap(PoolKey memory _key, bool zeroForOne, int256 amountSpecified, bytes memory hookData)
```

Comment:
wondering if it would be helpful to add native input support too? either a different function `swapNativeInput` or generalise this one a bit more.
Something like `bool isNativeInput = (zeroForOne && key.currency0 == NATIVE) ? true : false`
and then i think youd only have to change 2 things:
- `currencyAlreadySent: isNativeInput`
- `{ value: isNativeInput ? amountSpecified : 0 }`

Reply:
hm i guess that `value` formula only works if this is exact input not exact output... maybe it becomes too annoying to do this, what do you think?

Reply:
ah yeah. there's no reasonable way to cleanly abstract exact output with native token input

leaning towards a `swapNativeInput` vs overloading the simple one with an "optional" msgValue parameter.

Reply:
sgtm!